src/state/farms/hooks.ts

```diff
@@ -154,7 +154,7 @@ export const useFarmsData = () => {
     }
 
     checkForFarms()
-  }, [dispatch, ethPrice.currentPrice, chainId, fairLaunchContracts, account, blockNumber])
+  }, [apolloClient, dispatch, ethPrice.currentPrice, chainId, fairLaunchContracts, account, blockNumber, allTokens])
```

Comment:
earlier, when I added `allTokens` here, the component is infinite-loop because it's an array and it's not wrapped by useMemo. I decided to not include it in the dependency array because it doesn't change much.

please help to check if adding `allTokens` here will cause the component infinite-loop now, just for safety

Reply:
Checked. 
I've update all dependencies of other hooks which useAllTokens depend on, So It can be memorized